aioamqp/channel.py

```diff
@@ -488,15 +488,15 @@ def basic_publish(self, payload, exchange_name, routing_key, properties=None, ma
         header_frame = amqp_frame.AmqpRequest(
             self.protocol._stream_writer, amqp_constants.TYPE_HEADER, self.channel_id)
         header_frame.declare_class(amqp_constants.CLASS_BASIC)
-        header_frame.set_body_size(len(payload))
+        header_frame.set_body_size(payload_len)
         encoder = amqp_frame.AmqpEncoder()
         encoder.write_message_properties(properties)
         yield from self._write_frame(header_frame, encoder, drain=False)
 
         # split the payload
 
-        frame_max = self.protocol.server_frame_max or len(payload)
-        for chunk in (payload[0+i:frame_max+i] for i in range(0, len(payload), frame_max)):
+        frame_max = self.protocol.server_frame_max or payload_len
+        for chunk in (payload[0+i:frame_max+i] for i in range(0, payload_len, frame_max)):
```

Comment:
Not really related to your modifications but noticed maybe this line would benefit from an explicit function doing the chunking for clarity:
```python
def chunker(seq, size):                                
    for pos in range(0, len(seq), size):
        yield seq[pos:pos + size]
```
Testing the performance of this for various length of payloads yield similar results.
```
In [55]: frame_max = 10
In [56]: payload = ''.join([chr(i + 63) * 100 for i in range(60)])
In [57]: %timeit -n 10000 for chunk in chunker(payload, 10): print(chunk)
[...]
10000 loops, best of 3: 5.59 ms per loop
In [58]: %timeit -n 10000 for chunk in (payload[0+i:frame_max+i] for i in range(0, len(payload), frame_max)): print(chunk)
[...]
10000 loops, best of 3: 5.93 ms per loop
```

Reply:
Such long duration caused by huge I/O operation count because of print.
```python
def chunker2(seq, size):
    return [seq[i:size+i] for i in range(0, len(seq), size)]

def chunker(seq, size):
    for pos in range(0, len(seq), size):
        yield seq[pos:pos + size]
```
```
In [22]: %timeit -n 10000 for chunk in chunker(payload, 10): pass
10000 loops, best of 3: 107 µs per loop

In [23]: %timeit -n 10000 for chunk in chunker2(payload, 10): pass
10000 loops, best of 3: 96.2 µs per loop

In [24]: %timeit -n 10000 for chunk in (payload[i:frame_max+i] for i in range(0, len(payload), frame_max)): pass
10000 loops, best of 3: 127 µs per loop
```